text/0000-inferred-type-aliases.md

```diff
@@ -0,0 +1,696 @@
+   |
+11 | type Foo = _;
+   |            ^ not allowed in type signatures
+```
+
+This restriction is lifted by this RFC in `type` items.
+
+Instead, the `_` "type" is permitted within `type`s.
+However, each `type` item that uses `_` in any part of its definition
+must be constrained by at least one `fn` body, or `const`/`static` initializer
+(henceforth: *"determinant"*). All of these determinants must be in the same
+module as the `type` item inferred and must independently place constraints
+on the `type` item such that they determine the same underlying exact type for it.
+A body or initializer must either fully constrain or place no constraints upon a given `type` item with `_` inside it.
+
+A `type` item that uses `_` within its definition exposes its
+full type identity globally. In other words, `_` acts exactly like
+`existential type` (as specified in [RFC 2071]) does,
+with the exception of [transparency][transparent].
+
+Another difference between `_` in `type` and `existential type` is that
+the former may be used in `impl` blocks.
+
+[`E0277`]: https://doc.rust-lang.org/stable/error-index.html#E0277
+
+When a `<bound>` is present on `type <name> <generics>?: <bound> = <definition>`
+the type checker will check that `<definition>` satisfies the specified `<bound>`
+or will issue [`E0277`] otherwise.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The primary drawback to this feature is that can be *misused* for abstraction
+(which it should *not* be used for) instead of using the feature for what it
+was meant for (naming [voldemort types][voldemort type] and eliding
+unimportant implementation details).
+
+To mitigate this possibly, the use cases of using `_` should be highlighted
+and it should be stressed that the feature is not meant for abstraction purposes.
+The [guide-level-explanation] provides a series of [dos and don’ts][do_dont]
+that may be used in teaching material.
+
+Another possible mitigation strategy is to lint in the case of public aliases
+which include `_` in their definitions. However, this will not help for
+associated types.
+
+[semverver]: https://crates.io/crates/semverver
+
+Finally, another drawback is the somewhat increased risk of breaking changes
+if and when users mistake `_` as something [opaque] or if they are simply
+not careful enough when making changes. However, [interesting work][semverver]
+is being done in the area of detecting breaking changes. Such a tool could
+potentially also be used for detecting breakage where `_` is involved.
+
+# Rationale and alternatives
+[alternatives]: #rationale-and-alternatives
+
+## Use cases
+
+The use cases this RFC is intended to facilitate are discussed in the [motivation].
+It follows then that if this RFC is not accepted, those use cases will either
+have to be supported by some other means (such as `typeof`, see below)
+or they will remain unsupported.
+
+In particular, do note that this RFC enables things that are simply *impossible*
```

Comment:
Can you give an example of something that's impossible, or link back to a previous example, here? (As opposed to a user manually inputting the type that would have been inferred automatically.)

Reply:
Added a backlink to the motivation where this is detailed :)